dpctl/tensor/_utility_functions.py

```diff
@@ -0,0 +1,119 @@
+from numpy.core.numeric import normalize_axis_tuple
+
+import dpctl
+import dpctl.tensor as dpt
+import dpctl.tensor._tensor_impl as ti
+
+
+def _boolean_reduction(x, axis, keepdims, func):
+    if not isinstance(x, dpt.usm_ndarray):
+        raise TypeError(f"Expected dpctl.tensor.usm_ndarray, got {type(x)}")
+
+    nd = x.ndim
+    if axis is None:
+        axis = tuple(range(nd))
+    if not isinstance(axis, (tuple, list)):
+        axis = (axis,)
+    axis = normalize_axis_tuple(axis, nd, "axis")
+
+    exec_q = x.sycl_queue
+    res_usm_type = x.usm_type
+
+    red_nd = len(axis)
+    if red_nd == 0:
+        return dpt.astype(x, dpt.bool)
+
+    perm = [i for i in range(nd) if i not in axis] + list(axis)
+    x_tmp = dpt.permute_dims(x, perm)
+    res_shape = x_tmp.shape[: nd - red_nd]
```

Comment:
I would perhaps treat `axis is None` case differently (no need to call `normalize_axis_tuple`, we know `res_shape = tuple()`, `res_nd=nd` and `x_tmp = x` (hence to call to `permute_dims` is needed), and `perm = list(range(nd))`. 

I bet this will make `axis=None` case a little faster too.